### PR TITLE
Chore/thread manager

### PR DESCRIPTION
**Pull request by markiing**
- Creating abstraction for system thread (_4RThread), this thread will manage all thread operations (Start, suspend, creation, sleeps...)
- Marking all models uses _4RThread abstraction
- Removing unecessary Console.Writelines

### CODE REVIEW

File: Utils/_4RThread.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            this.thread = new Thread(() =>
+            {
+                while (true)
+                {
+                    try
+                    {
+                        toRun(0);
+                    }catch(Exception ex) { 
+                        Console.WriteLine("[4RThread Exception] Error while Executing Thread Method ==== "+ex.Message);
```
Comment (afonsocdaniel) [inline Utils/_4RThread.cs:24]:
```suggestion
                        Console.WriteLine("[4RThread Exception] Error while Executing Thread Method ==== " + 
 ex.Message);
```

File: Utils/_4RThread.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        public _4RThread(Func<int, int> toRun)
+        {
+            this.thread = new Thread(() =>
+            {
+                while (true)
+                {
+                    try
+                    {
+                        toRun(0);
+                    }catch(Exception ex) {
```
Comment (afonsocdaniel) [inline Utils/_4RThread.cs:23]:
```suggestion
                    } catch (Exception ex) { 
```

File: Model/Macro.cs [hunk 3]
```diff
@@ -75,53 +75,44 @@ public string GetConfiguration()
...
             {
-                this.macroThread.Abort();
+                this.thread = new _4RThread((_) => MacroExecutionThread(roClient));
+                _4RThread.Start(this.thread);
             }
         }
+
+        public void Stop()
+        {
+            _4RThread.Stop(this.thread);
```
Comment (afonsocdaniel) [inline Model/Macro.cs:115]:
```suggestion
            this._4RThread.Stop(this.thread);
```

File: Model/Macro.cs [hunk 4]
```diff
@@ -75,53 +75,44 @@ public string GetConfiguration()
...
+        public void Start()
         {
-            if (this.macroThread != null && this.macroThread.IsAlive)
+            Stop();
+            Client roClient = ClientSingleton.GetClient();
+            if (roClient != null)
             {
-                this.macroThread.Abort();
+                this.thread = new _4RThread((_) => MacroExecutionThread(roClient));
+                _4RThread.Start(this.thread);
```
Comment (afonsocdaniel) [inline Model/Macro.cs:109]:
```suggestion
                this._4RThread.Start(this.thread);
```

Comment (markiing) [reply Model/Macro.cs:109]: this is not a instance attribute. 
`_4RThread` is a class with static `Start` method.

File: Utils/_4RThread.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        public static void Stop(_4RThread _4RThread)
+        {
+            if (_4RThread != null && _4RThread.thread.IsAlive)
+            {
+                try
+                {
+                    _4RThread.thread.Suspend();
+                }
+                catch (Exception ex) {
+                    Console.WriteLine("[4R Thread Exception] =========== We could not suspend curren thread: " + ex);
```
[the remark this answers is not in the document]
Comment (markiing) [reply Utils/_4RThread.cs:49]: this is not a instance attribute.
_4RThread is a class with static Start method.

**Pull request merged**